compiler/p/codegen/ControlFlowEvaluator.cpp

```diff
@@ -3778,7 +3778,7 @@ static void lookupScheme2(TR::Node *node, bool unbalanced, bool fromTableEval, T
       {
       if (two_reg)
          {
-         loadConstant(cg, node, (int32_t)(preInt >> 32), valRegister->getHighOrder());
+         loadConstant(cg, node, (int32_t)((int64_t)preInt >> 32), valRegister->getHighOrder());
          loadConstant(cg, node, (int32_t)preInt, valRegister->getLowOrder());
```

Comment:
The fact that the declared type of `preInt` is `int32_t` but there are explicit casts of it to `int32_t` is a clue that there's some history here.  It turns out that under W-Code support &mdash; which has largely been removed &mdash;, `getCaseConstant` returned a value of type `int64_t`.  In addition, `isInt64`, which is always [set false above](https://github.com/dylanjtuttle/omr/blob/f809e80cf13fc95824106aa328e0d87c3f5dc112/compiler/p/codegen/ControlFlowEvaluator.cpp#L3730), could only be set true if W-Code support was enabled.

If you take a look at a similar piece of code that appears in [`lookupScheme1`](https://github.com/dylanjtuttle/omr/blob/f809e80cf13fc95824106aa328e0d87c3f5dc112/compiler/p/codegen/ControlFlowEvaluator.cpp#L3677-L3694), under the `isInt64` case, the result of a call to `getCaseConstant` is assigned to an `int64_t` variable.  I think that doing something similar here would be better.

Alternatively, maybe all the `isInt64` code could be removed &mdash; though that's a more drastic bit of clean-up.

Julian @zl-wang, thoughts?

Reply:
i would suggest removing those dead code in this routine.  it is relatively safer, considering the overloaded routine **loadConstant** (two different implementations: 64bit and 32bit, and later fanning out to call different things depending on JIT/AOT/SVM etc etc.  that is also why a cast was there i think).

Reply:
Should I just remove the dead code / refactor the else block in [this if statement](https://github.com/dylanjtuttle/omr/blob/f809e80cf13fc95824106aa328e0d87c3f5dc112/compiler/p/codegen/ControlFlowEvaluator.cpp#L3777-L3786)? There are two other places in this function with `isInt64` related logic, as well as a few other functions (the one exception is [here](https://github.com/dylanjtuttle/omr/blob/f809e80cf13fc95824106aa328e0d87c3f5dc112/compiler/p/codegen/ControlFlowEvaluator.cpp#L4632) where we probe a `TR::DataType` variable to see if it `isInt64`). Should I clean those up as well?

Reply:
don't get too ambitious.  limit the clean-up to this **lookupScheme2** routine this time around.